libraries/AC_Fence/AC_PolyFence_loader.cpp

```diff
@@ -39,72 +197,1270 @@ bool AC_PolyFence_loader::load_point_from_eeprom(uint16_t i, Vector2l& point)
+                break;
+            }
+            _num_loaded_inclusion_boundaries++;
+            break;
+        }
+        case AC_PolyFenceType::POLYGON_EXCLUSION: {
+            ExclusionBoundary &boundary = _loaded_exclusion_boundary[_num_loaded_exclusion_boundaries];
+            boundary.points = next_storage_point;
+            boundary.count = index.count;
+            storage_offset += 1; // skip vertex count
+            if (!read_polygon_from_storage(ekf_origin, storage_offset, index.count, next_storage_point)) {
+                gcs().send_text(MAV_SEVERITY_WARNING, "AC_Fence: polygon read failed");
+                storage_valid = false;
+                break;
+            }
+            _num_loaded_exclusion_boundaries++;
+            break;
+        }
+        case AC_PolyFenceType::CIRCLE_EXCLUSION: {
+            ExclusionCircle &circle = _loaded_circle_exclusion_boundary[_num_loaded_circle_exclusion_boundaries];
+            if (!read_scaled_latlon_from_storage(ekf_origin, storage_offset, circle.loc)) {
+                gcs().send_text(MAV_SEVERITY_WARNING, "AC_Fence: latlon read failed");
+                storage_valid = false;
+                break;
+            }
+            // now read the radius
+            circle.radius = fence_storage.read_uint32(storage_offset);
+            _num_loaded_circle_exclusion_boundaries++;
+            break;
+        }
+        case AC_PolyFenceType::CIRCLE_INCLUSION: {
+            InclusionCircle &circle = _loaded_circle_inclusion_boundary[_num_loaded_circle_inclusion_boundaries];
+            if (!read_scaled_latlon_from_storage(ekf_origin, storage_offset, circle.loc)) {
+                gcs().send_text(MAV_SEVERITY_WARNING, "AC_Fence: latlon read failed");
+                storage_valid = false;
+                break;
+            }
+            // now read the radius
+            circle.radius = fence_storage.read_uint32(storage_offset);
+            _num_loaded_circle_inclusion_boundaries++;
+            break;
+        }
+        case AC_PolyFenceType::RETURN_POINT:
+            if (_loaded_return_point != nullptr) {
+                gcs().send_text(MAV_SEVERITY_WARNING, "PolyFence: Multiple return points found");
+                storage_valid = false;
+                break;
+            }
+            _loaded_return_point = next_storage_point;
+            if (!read_scaled_latlon_from_storage(ekf_origin, storage_offset, *next_storage_point)) {
+                storage_valid = false;
+                gcs().send_text(MAV_SEVERITY_WARNING, "PolyFence: latlon read failed");
+                break;
+            }
+            next_storage_point++;
+            break;
+        }
+    }
+
+    if (!storage_valid) {
+        unload();
+        return false;
+    }
+
+    _load_time_ms = AP_HAL::millis();
+
+    return true;
+}
+
+Vector2f* AC_PolyFence_loader::get_exclusion_polygon(uint16_t index, uint16_t &num_points) const
+{
+    if (index > _num_loaded_exclusion_boundaries) {
```

Comment:
Should this be ">="?  If "yes" then I think the same issue may be present in the get_ functions for other exclusion/inclusion zones.

Reply:
Fixed.